libgalois/src/analytics/local_clustering_coefficient/local_clustering_coefficient.cpp

```diff
@@ -0,0 +1,287 @@
+      katana::loopname("TriangleCount_OrderedCountAlgo"));
+
+  katana::do_all(
+      katana::iterate(*graph),
+      [&](const Node& n) {
+        for (uint32_t i = 0; i < num_threads; i++) {
+          graph->GetData<NodeTriangleCount>(n) +=
+              (*(per_thread_node_triangle_count.getRemote(i)))[n];
+        }
+      },
+      katana::chunk_size<kChunkSize>(), katana::steal(),
+      katana::loopname("TriangleCount_Reduce"));
+}
+
+void
+ComputeLocalClusteringCoefficient(Graph* graph) {
+  katana::do_all(katana::iterate(*graph), [&](Node n) {
+    auto degree = std::distance(graph->edges(n).begin(), graph->edges(n).end());
+    graph->template GetData<NodeClusteringCoefficient>(n) =
+        ((double)(2 * graph->template GetData<NodeTriangleCount>(n))) /
+        (degree * (degree - 1));
+  });
+
+  return;
+}
+}  // namespace
+
+katana::Result<void>
+katana::analytics::LocalClusteringCoefficient(
+    katana::PropertyGraph* pg, const std::string& output_property_name,
+    LocalClusteringCoefficientPlan plan) {
+  katana::StatTimer timer_graph_read(
+      "GraphReadingTime", "LocalClusteringCoefficient");
+  katana::StatTimer timer_auto_algo(
+      "AutoRelabel", "LocalClusteringCoefficient");
+
+  bool relabel;
+  timer_graph_read.start();
+  switch (plan.relabeling()) {
+  case LocalClusteringCoefficientPlan::kNoRelabel:
+    relabel = false;
+    break;
+  case LocalClusteringCoefficientPlan::kRelabel:
+    relabel = true;
+    break;
+  case LocalClusteringCoefficientPlan::kAutoRelabel:
+    timer_auto_algo.start();
+    relabel = IsApproximateDegreeDistributionPowerLaw(*pg);
+    timer_auto_algo.stop();
+    break;
+  default:
+    return katana::ErrorCode::AssertionFailed;
+  }
+
+  std::unique_ptr<katana::PropertyGraph> mutable_pfg;
+  if (relabel || !plan.edges_sorted()) {
+    // Copy the graph so we don't mutate the users graph.
+    auto mutable_pfg_result = pg->Copy({}, {});
+    if (!mutable_pfg_result) {
+      return mutable_pfg_result.error();
+    }
+    mutable_pfg = std::move(mutable_pfg_result.value());
+    pg = mutable_pfg.get();
+  }
+
+  if (relabel) {
+    katana::StatTimer timer_relabel(
+        "GraphRelabelTimer", "LocalClusteringCoefficient");
+    timer_relabel.start();
+    if (auto r = katana::SortNodesByDegree(pg); !r) {
+      return r.error();
+    }
+    timer_relabel.stop();
+  }
+
+  // If we relabel we must also sort. Relabeling will break the sorting.
+  if (relabel || !plan.edges_sorted()) {
+    if (auto r = katana::SortAllEdgesByDest(pg); !r) {
+      return r.error();
+    }
+  }
+
+  timer_graph_read.stop();
+
+  katana::Prealloc(1, 16 * (pg->num_nodes() + pg->num_edges()));
+  katana::reportPageAlloc("LCC_MeminfoPre");
+
+  katana::analytics::TemporaryPropertyGuard temporary_property{pg};
+
+  if (auto result = katana::analytics::ConstructNodeProperties<NodeData>(
+          pg, {output_property_name, temporary_property.name()});
```

Comment:
We don't need temporary_property for atomicAlgo. 

I would suggest making these algorithms into separate structs with their own NodeData and EdgeData types ComputeTriagles and ComputeCoefficient functions. See connected-components for example.

Reply:
ok, I will do that then